services/admissioncontrol/admission_policy/admission_control_policy.go

```diff
@@ -0,0 +1,86 @@
+package admission_policy
+
+import (
+	"fmt"
+	"net/http"
+)
+
+const (
+	admissionControlPolicyResourcePath = "kubernetes/admissionControl/policy"
+)
+
+type AdmissionControlPolicyRequest struct {
+	TargetId        string   `json:"targetId"`
+	TargetType      string   `json:"targetType,omitempty"`
+	RulesetId       int      `json:"rulesetId"`
+	NotificationIds []string `json:"notificationIds"`
+	Action          string   `json:"action"`
+	RulesetPlatform string   `json:"rulesetPlatform"`
+}
+
+type AdmissionControlPolicyResponse struct {
+	ID              string   `json:"id"`
+	TargetId        string   `json:"targetId"`
+	TargetType      string   `json:"targetType"`
+	RulesetId       int      `json:"rulesetId"`
+	Action          string   `json:"action"`
+	NotificationIds []string `json:"notificationIds"`
+	ErrorMessage    string   `json:"errorMessage"`
+}
+
+func (service *Service) GetAdmissionControlPolicy(id string) (*AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new(AdmissionControlPolicyResponse)
+	path := fmt.Sprintf("%s/%s", admissionControlPolicyResourcePath, id)
+	resp, err := service.Client.NewRequestDo("GET", path, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) GetAllAdmissionControlPolicies() (*[]AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new([]AdmissionControlPolicyResponse)
+	resp, err := service.Client.NewRequestDo("GET", admissionControlPolicyResourcePath, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) Create(body *AdmissionControlPolicyRequest) (*AdmissionControlPolicyResponse, *http.Response, error) {
+	v := new([]AdmissionControlPolicyResponse)
+	resp, err := service.Client.NewRequestDo("POST", admissionControlPolicyResourcePath, nil, []*AdmissionControlPolicyRequest{body}, v)
+	if err != nil {
+		return nil, nil, err
+	}
+	policy := new(AdmissionControlPolicyResponse)
+	if len(*v) > 0 {
```

Comment:
here we check if we get list with element numbers greater than 0, if yes we get the policy struct from the response after json decoding it so we got the policy object from it , also this is because our api return back list of AdmissionControlPolicyRequestModel , this make this api return only the created policy

Reply:
why should this code be different from the API (and not return all)? to my mind this should be the same as the API itself. (if we don't want this behavior we should fix it in the API) looking at the same. again looking at the example I gave above I don't see this being done

Reply:
this behaviour is really similar to this one :  https://github.com/dome9/dome9-sdk-go/blob/master/services/compliance/continuous_compliance_policy/continuous_compliance_policy.go#L58 , so it should be similar as our api return list and we took the required element from the response